types/index.d.ts

```diff
@@ -907,6 +908,16 @@ declare namespace Knex {
   interface Distinct<TRecord extends {}, TResult = {}[]>
     extends ColumnNameQueryBuilder<TRecord, TResult> {}
 
+  interface DistinctOn<TRecord extends {}, TResult = {}[]> {
+    <ColNameUT extends keyof TRecord>(
+      ...columnNames: readonly ColNameUT[]
+    ): QueryBuilder<TRecord, TResult>;
+
+    <ColNameUT extends keyof TRecord>(
+      columnNames: readonly ColNameUT[]
+    ): QueryBuilder<TRecord, TResult>;
+  }
```

Comment:
It is also good to add another signature with `columnNames: readonly string[]` to accomodate things like `distinctOn('users.foo')` where the argument is not the same as column key.